File: src/main/java/de/aquafun3d/vanilla/listeners/BackpackListener.java
package de.aquafun3d.vanilla.listeners;

import de.aquafun3d.vanilla.Main;
import net.kyori.adventure.text.Component;
import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryCloseEvent;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;

import java.io.IOException;

public class BackpackListener implements Listener {
	@EventHandler
	public void onInvClose(InventoryCloseEvent e) throws IOException {
		Component title = e.getView().title();
		Inventory inv = e.getInventory();
		Player player = (Player) e.getPlayer();
		if(title.equals(Component.text(ChatColor.DARK_PURPLE + "Backpack"))) {
			Main.vanillaConfig.set("backpack." + player.getUniqueId(), null);
			for(int i = 0; i < 27;i++){
				ItemStack item = inv.getItem(i);
				if(item == null){
					continue;
				}
				if(item.getType() == Material.AIR){
					continue;
				}
				Main.vanillaConfig.set("backpack." + player.getUniqueId() + "." + i, item);
			}
		}
	}
}

File: src/main/kotlin/de/aquafun3d/vanilla/listeners/BackpackListener.kt
package de.aquafun3d.vanilla.listeners

import de.aquafun3d.vanilla.Main
import net.kyori.adventure.text.Component
import org.bukkit.ChatColor
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryCloseEvent
import java.awt.TextComponent
import java.io.IOException

class BackpackListener: Listener {

	@EventHandler
	@Throws(IOException::class)
	fun onInvClose(e: InventoryCloseEvent) {
		val title = e.view.title()
		val inv = e.inventory
		val player = e.player as Player
		if (title == Component.text(ChatColor.DARK_PURPLE.toString() + "Backpack")) {
			Main.vanillaConfig?.set("backpack." + player.uniqueId.toString(), null)
			for (i in 0..26) {
				val item = inv.getItem(i) ?: continue
				if (item.type == Material.AIR) continue
				Main.vanillaConfig?.set("backpack." + player.uniqueId.toString() + "." + i, item)
			}
		}
	}
}